### PR TITLE
updates

### LINKED ISSUE #25: Ionic 3 Please add a @NgModule annotation Error
Hi 
i love this package. but it is not running with ionic 3 i got this error:
Please add a @NgModule annotation

Can you fix this?

Thanks,
Daniel

### PR DESCRIPTION
**Pull request by ihadeed**
- removed dependencies
  - made everything a dev dependency to make things simpler
- added `NgModule` ( closes #25  )
- added .editorconfig file
- removed typings and added `@types`
- compiling with `ngc` instead of `tsc` to produce needed files for AoT compilation
- added tests
- fix linting
- add circleci instructions

I might add more things to this PR .. still testing the library.. these are just the first things I noticed that need changing

### CODE REVIEW

**Comment by Nodonisko** [comment]
Wow! Such a nice PR, I think we should publish this as version 2.0.0

File: src/cache.service.ts [hunk 1]
```diff
@@ -1,23 +1,31 @@
 import { Injectable } from '@angular/core';
 import { SqlStorage } from './storage';
-import { Observable, Subject } from 'rxjs/Rx';
+import { Observable } from 'rxjs/Observable';
+import { Subject } from 'rxjs/Subject';
 import { Request, Response, ResponseOptions } from '@angular/http';
-
-const MESSAGES = {
+import 'rxjs/add/observable/fromPromise';
```
Comment (Nodonisko) [inline src/cache.service.ts]: Is this imports really necessary? I think this operators is already hidden under rxjs/Observable

File: .gitignore [hunk 2]
```diff
@@ -1,14 +1,5 @@
-# Created by .ignore support plugin (hsz.mobi)
```
Comment (Nodonisko) [inline .gitignore]: I think this should stay here

**Comment by Nodonisko** [comment]
It looks really nice, please fix .gitignore and we can merge it :)

**Comment by ihadeed** [comment]
Will update it soon, just finishing up the tests.

**Comment by ihadeed** [comment]
Ok @Nodonisko all is done!

I also added CircleCI configuration to the project. You can see a sample build here: https://circleci.com/gh/ihadeed/ionic-cache/1

You just need to sign up there and add your repo so it gets built automatically on pushes/PRs. It's free for open source projects.

**Comment by Nodonisko** [comment]
It's really awesome. I will merge it and after I will fix few more issues and we can release.

Thanks very much! I will mention you in readme :)

**Comment by ihadeed** [comment]
It's worth noting that there's one breaking change. I renamed `disableCache` to `enableCache`. I thought it would be more intuitive and straightforward. 

So to disable cache you need to call:
```
this.cache.enableCache(false);
```

and to re-enable you call:
```
this.cache.enableCache(true);
```

**Comment by Nodonisko** [comment]
Yes, I think so, it was really bad naming.

But best change is tests, it will help a lot.

**Pull request merged**